src/meta/src/hummock/manager/mod.rs

```diff
@@ -732,6 +733,11 @@ where
         compaction_group_id: CompactionGroupId,
         selector: &mut Box<dyn LevelSelector>,
     ) -> Result<Option<CompactTask>> {
+        // TODO: `get_all_table_options` will hold catalog_manager async lock, to avoid holding the
```

Comment:
Prior to this PR, we held catalog_manager lock in compaction_guard on `get_table_options`. Why can't we do the same things? Potential deadlock found?

Reply:
nvm. Saw the review comment above.